src/utils/use-focus-on-load.ts

```diff
@@ -38,15 +39,21 @@ export function useFocusOnLoad(setFocusOnLoad: boolean = true): Ref<any> {
   return focusCallbackRef;
 }
 
-export const giveFocus = (
-  focusRef?: React.RefObject<any> | null | undefined,
-) => {
+type focusRefType = React.RefObject<any> | null | undefined;
+
+const focusOnRef = (focusRef: focusRefType) => {
+  const reactTag = findNodeHandle(focusRef?.current);
+  reactTag && AccessibilityInfo.setAccessibilityFocus(reactTag);
+};
+
+export const giveFocus = (focusRef?: focusRefType) => {
   if (focusRef && focusRef.current) {
     InteractionManager.runAfterInteractions(() => {
-      setTimeout(() => {
-        const reactTag = findNodeHandle(focusRef?.current);
-        reactTag && AccessibilityInfo.setAccessibilityFocus(reactTag);
-      }, 100);
+      if (Platform.OS === 'ios') {
+        focusOnRef(focusRef);
+      } else {
+        setTimeout(() => focusOnRef(focusRef), 100);
```

Comment:
I know is not ready yet but would be nice to invalidate the timer in case of a user go back or something else, before the timer is fired otherwise will create an issue. It looks like this is impossible during the 100ms delay, but might be safe to call `clearInterval(intervalId)`

Reply:
I like this suggestion!